### PR TITLE
fix: add optional flag to generate at_demo_data.dart script and skip …

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
* Fix the failing e2e tests. 
* Please refer to the run: https://github.com/atsign-foundation/at_server/actions/runs/4559428003 which has all the tests pass successfully.

**- How I did it**
* Added an optional flag `-p pkam` to the `dart bin/generate_at_demo_data.dart` command.

**- How to verify it**
* All the other tests are running successfully.

### CODE REVIEW

**Review by gkc — CHANGES_REQUESTED** [review]
We shouldn't mark the test as skipped; however, by design right now, we don't want to be able to switch on testingMode on these new servers. Can you instead find a way to skip it just in this situation? Could base it on an initial check to see if testingMode  is enabled or not.

**Review by kumarnarendra701 — APPROVED** [review]

**Comment by sitaram-kalluri** [comment]
> We shouldn't mark the test as skipped; however, by design right now, we don't want to be able to switch on testingMode on these new servers. Can you instead find a way to skip it just in this situation? Could base it on an initial check to see if testingMode is enabled or not.

@gkc: Modified the test which checks if the testing mode is enabled on the secondary server. Test executes only if enabled, else skips.

Below is the test result of the bypass cache test when the testing mode is disabled:
```
test/cached_key_test.dart: update-lookup verb passing bypassCache 
  Clearing socket response queues
  command sent: config:print:autoNotify
  Response: data:testing mode disabled by default
  Testing mode is not enabled on the secondary server: @test7bbaa28e_5bd5_4332_97fd_717b868d77901. So skipping the test
```

File: tests/at_end2end_test/test/cached_key_test.dart [hunk 1]
```diff
@@ -143,12 +142,22 @@ void main() {
   try {
     test('update-lookup verb passing bypassCache ', () async {
       ///Update verb on atsign_1
+      // Check if TestingMode is enabled on the secondary server
+      // If yes allow the test to run, else skip
+      await sh1.writeCommand('config:print:autoNotify\n');
+      String response = await sh1.read();
+      // If the response is not true, testingMode is NOT enabled on the secondary server
+      // So, do not run the test.
+      if (!response.contains('true')) {
```
Comment (gkc) [inline tests/at_end2end_test/test/cached_key_test.dart]: A more exact comparison would be better. Ideally an equality something like `if (!response == 'data:true')` and just for completeness compare with `data:false` also

File: .github/workflows/e2e_test.yaml [hunk 2]
```diff
@@ -201,3 +201,19 @@ jobs:
...
+      # Verify if Atsign exists
+      - name: Verify if Atsign exists
+        run: |  
+          sleep 20
+          AT_SIGN_1_HOST_RESP=$((echo $AT_SIGN_1; sleep 1) | openssl s_client -connect root.atsign.wtf:64 2>/dev/null | grep --color=none "^@.*:" | cut -d'@' -f2)
+          AT_SIGN_2_HOST_RESP=$((echo $AT_SIGN_2; sleep 1) | openssl s_client -connect root.atsign.wtf:64 2>/dev/null | grep --color=none "^@.*:" | cut -d'@' -f2)
+          if [ -z "$AT_SIGN_1_HOST_RESP" ] && [ -z "$AT_SIGN_2_HOST_RESP" ]; then
+            echo "Atsign deleted successfully"
+          else
+            echo "Atsign still exists"
```
Comment (gkc) [inline .github/workflows/e2e_test.yaml:214]: Would be nice to have it echo the atSigns in the message also. Makes it easier to debug if / when it happens

**Review by gkc — CHANGES_REQUESTED** [review]

File: tests/at_end2end_test/test/cached_key_test.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply tests/at_end2end_test/test/cached_key_test.dart]: Replaced the IF condition as you suggested. But when testingMode is disable, the is response is not `data:false`. We get the following response. So not adding the check with `data:false`

```
@sitaram@config:print:autoNotify
data:testing mode disabled by default
```

File: .github/workflows/e2e_test.yaml (hunk 2 above)
Comment (sitaram-kalluri) [reply .github/workflows/e2e_test.yaml:214]: @kumarnarendra701: Can you please take care of this?

File: tests/at_end2end_test/test/cached_key_test.dart (hunk 1 above)
Comment (gkc) [reply tests/at_end2end_test/test/cached_key_test.dart]: My point was that autoNotify might be set to false ... i.e. if testingMode is enabled, you might get a response of true or false; but if testingMode is disabled, you get a response like "testing mode disabled by default"

Comment (sitaram-kalluri) [reply tests/at_end2end_test/test/cached_key_test.dart]: Got it Gary, Updated the IF condition and pushed the changes.

**Review by gkc — APPROVED** [review]

**Pull request merged**